source/_posts/2015-10-11-a-configuration-management-rosetta-stone-configuring-sensu-with-puppet.markdown

```diff
@@ -0,0 +1,352 @@
+config file or anything, but instead can only be added by special invocations
+of the `rabbitmqctl` command.
+
+Puppet strongly emphasizes code-reuse. The [Puppet Forge](https://forge.puppetlabs.com/) is the registry where you can upload and
+share modules. The Forge has a number of methods to help divine code quality.
+It also exposes "officially supported" and "officially approved" modules, for
+extra approval stamps. While the forge may have a very "long tail" of modules
+that do very common tasks, the set of officially-supported and
+officially-approved modules leaves behind a good selection of high-quality
+modules ready for re-use.
+
+#### Review of the Sensu Puppet Module
+
+The [sensu-puppet module](https://github.com/sensu/sensu-puppet) is a
+first-class citizen in the Sensu world. It has native types for the Sensu JSON
+files that it manages, as well as a `sensu-gem` type for easily installing
+rubygems with the embedded Sensu ruby.
+
+The Sensu Puppet module only manages Sensu, and has no integration with any
+other RabbitMQ, Redis, or any other module. To me this is expected, in the
+Puppet world it would be the job of a `profile` to combine the Sensu module
+with RabbitMQ and other things. For the most part this integration is left as
+an exercise to the reader.
+
+The Sensu Puppet module also doesn't manage Uchiwa. That requires a
+[different puppet module](https://github.com/Yelp/puppet-uchiwa). Again to me
+this is a good thing, I hate it when tools try to do too much.
+
+The actual codebase is actively maintained and reasonably active, with a few
+releases per year. The Puppet Forge has almost perfect scores for it's quality
+ratings. The code has excellent unit test and acceptance test coverage. As far
+as puppet modules go, the Sensu puppet module is a great example of a
+well-maintained piece of code.
+
+One downside the "completeness" of the module is that sometimes new features of
+Sensu are released, and the puppet-module will lag. The configuration inputs to
+the puppet module are well-typed, and not just free-form hashes. This gives a
+lot of guard-rails and helps ensure config files are correct before they hit
+the disk, but it means that some features are not usable until the Puppet
+module can account for them.
+
+Although the code worked, there was a [significant bug](https://github.com/sensu/sensu-puppet/issues/336) that prevented the
+module from ever converging. This was annoying but allowed me to test the code.
+This bug looks to be fixed in master.
+
+### Chef
+
+#### Chef in General
+
+Chef is not as old as Puppet, but is certainly a mature product. Chef is "just
+ruby" when it comes to it's configuration language. The upside to this is that
+Ruby developers can theortically dive in and hack on stuff. The downside to
+this is that being "just ruby", "leaves a lot of rope to hang yourself".
+
+One nice feature provided by the Chef-company is their hosted-chef solution, which
```

Comment:
I would remove most of the hyphenation.  "Chef company is their hosted Chef solution"